### PR TITLE
Migration to QGIS 3

### PR DESCRIPTION
**Pull request by lbartoletti**
This migration consists to:

- Drop .ui forms to drag n' drop forms.
- Update symbols rules
- Update macros
- Some improvements about alt geometry and drop useless stuff thanks to the powerfull Qgis 3.

### CODE REVIEW

**Comment by haubourg** [comment]
of course +1 on my side. 
@ponceta @kandre Any test on your side is welcome

[funded by Charente Eaux]

**Comment by lbartoletti** [comment]
@ponceta @kandre Just a gentle reminder to update QGIS project to v3 :wink:

**Comment by haubourg** [comment]
A little Up here. Anyone willing to test the QGIS3 version of this public project?

**Comment by ponceta** [comment]
I made a couple of tests, this works fine! Thanks for the update!

image

**Comment by kandre** [comment]
Great, thanks for this update ! 
Regarding commits named "remove useless ui file...", do you suggest to abandon gently the ui form files to dragndrop ones in general with QGIS 3 ?

**Comment by lbartoletti** [comment]
> Regarding commits named "remove useless ui file...", do you suggest to abandon gently the ui form files to dragndrop ones in general with QGIS 3 ?

As possible, this simplifies projects IMHO

**Comment by haubourg** [comment]
@kandre in fact there should be no more .ui file for forms. It's a lot simpler to maintain and QWAT is easier for newcomers. 
On UI is still there for the alternative geometry dialog chooser.

**Comment by haubourg** [comment]
> There's something strange with the use of chained filters and update but that's probably a known issue?

@ponceta can you clarify what is wrong ?

**Comment by kandre** [comment]
I have some new questions about this PR : 
- "[switch to trackable project file](https://github.com/qwat/QWAT/pull/299/commits/6b4a392c3bd832dd6a6777947fbfc438e79aa934)" : what does mean trackable project file ?
- "[add valve: missing checkbox for geometry alternative](https://github.com/qwat/QWAT/pull/299/commits/0d2f6846626f752b4d251f91dfae053acf3af1d5)" : I haven't found the checkbox, where is it placed ?
- "[clean layer actions](https://github.com/qwat/QWAT/pull/299/commits/d23df68782babbb290ed88fd22769a6cae90a746)": if one clean those actions, does the postgresql audit used for data changes history still working ?

**Comment by haubourg** [comment]
> "[switch to trackable project file](https://github.com/qwat/QWAT/pull/299/commits/6b4a392c3bd832dd6a6777947fbfc438e79aa934)" : what does mean trackable project file ?
 TLDR, it is is of no use currently.
It means we use the plugin https://github.com/opengisch/qgis_trackable_project_files/ to (try to) write the xml file in a more reproductible way so that git diff are usable. In fact, it does not work correctly yet since it works for one machine, but does not serialize the same way on another machine. I think it also requires a more recent version of QGIS, and is still a work in progress.



>     * "[add valve: missing checkbox for geometry alternative](https://github.com/qwat/QWAT/pull/299/commits/0d2f6846626f752b4d251f91dfae053acf3af1d5)" : I haven't found the checkbox, where is it placed ?

It should be in the dialog poping up when you start an edit session. I will raise only if you enable macros. It is the feature that allow you to choose if you want to edit both geometries or only the detailed geometry of a pipe. 

>     * "[clean layer actions](https://github.com/qwat/QWAT/pull/299/commits/d23df68782babbb290ed88fd22769a6cae90a746)": if one clean those actions, does the postgresql audit used for data changes history still working ?

Yes, those were automatically added by the old plugin Pg-audit_91 but are of no use currently, all is done on the DB side.

**Comment by ponceta** [comment]
> @ponceta can you clarify what is wrong ?

I've opened an issue here, since this is QGIS related.

https://git.oslandia.net/Support-projects/helpdesk_qwat_qgep/issues/86

**Comment by lbartoletti** [comment]
@ponceta Can we merge it?

**Comment by ponceta** [comment]
Yes! Please!

**Comment by haubourg** [comment]
@lbartoletti I let you the honor of pressing the green button :)

**Comment by lbartoletti** [comment]
> @lbartoletti I let you the honor of pressing the green button :)

:boom::boom::boom:

**Comment by lbartoletti** [comment]
Done :)

Thanks

**Comment by ponceta** [comment]
QWAT is now officially QGIS 3.X Ready! Would be a nice thing to put on the qwat news! I'll check that with Amandine.
